experimental/metal/direct_command_buffer.m

```diff
@@ -393,6 +394,10 @@ static void iree_hal_metal_command_buffer_destroy(iree_hal_command_buffer_t* bas
 
   iree_hal_metal_command_buffer_reset(command_buffer);
 
+  // Decrease command buffer refcount in the shared staging buffer, and potentially reclaim
+  // resources. We tie this to the command buffer's lifetime to avoid resource leak.
+  iree_hal_metal_staging_buffer_decrease_refcount(command_buffer->staging_buffer);
```

Comment:
unless decrease_refcount handles NULL you'll want to NULL guard this in cases where the creation failed partway through